client/verta/tests/unit_tests/pipeline/test_registered_pipeline.py

```diff
@@ -0,0 +1,206 @@
+        },
+        status=200,
+    )
+    # Upload the artifact
+    mocked_responses.put(
+        f"https://account.s3.amazonaws.com/development/ModelVersionEntity/{pipeline.id}/pipeline.json",
+        json={},
+        status=200,
+    )
+    pipeline._log_pipeline_definition_artifact()
+
+
+def test_to_pipeline_definition(
+    make_mock_pipeline_graph, make_mock_registered_model_version
+) -> None:
+    """
+    Test that a pipeline definition can be constructed from a
+    RegisteredPipeline object.  In depth testing of the `_to_graph_definition`
+    and `to_steps_definition` functions are handled in unit tests for
+    PipelineGraph.
+    """
+    graph = make_mock_pipeline_graph()
+    pipeline = RegisteredPipeline(
+        pipeline_graph=graph,
+        registered_model_version=make_mock_registered_model_version(),
+    )
+    pipeline_definition = pipeline._to_pipeline_definition()
+    assert pipeline_definition == {
+        "pipeline_version_id": pipeline.id,
+        "graph": graph._to_graph_definition(),
+        "predecessors": graph._to_steps_definition(),
+    }
+
+
+def test_to_pipeline_configuration_valid(
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+    make_mock_step_resources,
+) -> None:
+    """
+    Test that a valid pipeline configuration can be constructed from a
+    RegisteredPipeline object and a valid list of pipeline resources.
+    """
+    graph = make_mock_pipeline_graph()
+    step_names = [step.name for step in graph.steps]
+    mock_res = make_mock_step_resources(step_names)
+    pipeline = RegisteredPipeline(
+        pipeline_graph=graph,
+        registered_model_version=make_mock_registered_model_version(),
+    )
+
+    pipeline_configuration = pipeline._to_pipeline_configuration(
+        pipeline_resources=mock_res
+    )
+    assert pipeline_configuration["pipeline_version_id"] == pipeline.id
+    for graph_step, config_step in zip(graph.steps, pipeline_configuration["steps"]):
+        # All steps are included in the configuration
+        assert graph_step.name == config_step["name"]
+        # All steps in the config have resources
+        assert "resources" in config_step.keys()
+
+
+def test_to_pipeline_configuration_invalid_resources(
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+    make_mock_step_resources,
+) -> None:
+    """
+    Test that a ValueError is raised when an invalid step name is included
+    in the provided pipeline resources. (Does not match a step name in the
+    pipeline's graph)
+    """
+    graph = make_mock_pipeline_graph()
+    step_names = [step.name for step in graph.steps]
+    mock_res = make_mock_step_resources(step_names)
+    mock_res["invalid_step_name"] = make_mock_step_resources(["invalid_step_name"])
+    pipeline = RegisteredPipeline(
+        pipeline_graph=graph,
+        registered_model_version=make_mock_registered_model_version(),
+    )
+
+    with pytest.raises(ValueError):
```

Comment:
I'd recommend checking for the specific error message too if we can, since `_to_pipeline_configuration()` does _a lot_ of things, and `ValueError`s are rather common.

Reply:
Updated to check for this and more specific error conditions